Retrieve PHP version via API

AleTorrisi opened the pull request:
## Proposed changes

Use HUAPI API to retrieve and show PHP version in the Hosting panel.

## Type of Change

<!-- What types of changes does your code introduce? -->
<!-- _Put an `x` in the boxes that apply_ -->

#### Production

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update
- [ ] Refactoring / housekeeping (changes to files not directly related to functionality)

<!-- Bugfixes should explain how to reproduce the bug -->

#### Development

- [ ] Tests
- [ ] Dependency update
- [ ] Environment update / refactoring
- [ ] Documentation Update

<!-- All PRs should endeavor to include tests -->
<!-- PRs with new tools or dependencies should explain how to use them -->

## Video

<!-- Add a short video demonstrating where the change takes effect and how to can be reproduced by reviewers -->
<!-- On macOS press cmd+shift+5 to open the screen recording tool. It will save the video to the desktop  -->

<!-- Drag and drop the video file into the GitHub editor to attach it -->

## Checklist

<!-- _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._ -->

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] I have viewed my change in a web-browser
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

<!-- 
## Pre-deployment Checklist

- [ ] E.g. environmental variables which need to be set before deployment will work.
- [ ] E.g. other tickets which need to be complete before an API this change relies on will be ready
-->

<!--
## Post-deployment Checklist

How can the change be verified?

- [ ] E.g. temporarily enable logging and observe specific logs.
- [ ] E.g. observe new data in a specific view or table.
-->

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->
<!-- Now is a good time to create additional tickets for any necessary follow-up work. -->

AleTorrisi committed:
Tweak: use of API to retrieve SSH info

AleTorrisi committed:
Tweak: use API to get PHP version

AleTorrisi committed:
New: use HUAPI to retrieve PHP data

AleTorrisi committed:
Fix: usage of HUAPI response

AleTorrisi committed:
Fix: phpcs

arunshenoy99 commented on includes/SSHInfo/SSHInfo.php:
```diff
@@ -38,10 +38,35 @@ public function get_data() {
-		$ip       = $this->get_host_ip_from_hostname();
-		$username = $this->get_server_username();
-		$ssh_info = $username && $ip ? "{$username}@{$ip}" : '';
+		$hosting_id = 'xx';
+		$token      = 'xx';
+		// Inserted code for remote SSH info retrieval
+		$response = wp_remote_get(
+			"https://hosting.uapi.newfold.com/v1/hosting/{$hosting_id}/ssh",
+			array(
+				'headers' => array(
+					'Authorization' => 'Bearer ' . $token,
+				),
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
```
@AleTorrisi, let's update this code to use the HUAPIHelper as well. Thanks!

AleTorrisi committed:
Retrieve dynamically data using HUAPIHelper

arunshenoy99 commented on includes/SSHInfo/SSHInfo.php:
```diff
@@ -31,17 +41,39 @@ public function __construct( $container ) {
+		$response = $helper->send_request();
+
+		if ( is_wp_error( $response ) ) {
+			error_log( 'Error in the API call: ' . $response->get_error_message() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
```
Let's remove the `error_log`?

AleTorrisi committed:
Tweak: support for Atomic

AleTorrisi committed:
Fix: split check on is_atomic

dependabot[bot] committed:
NPM Dev(deps-dev): Bump @wordpress/scripts from 30.16.0 to 30.17.0

Bumps [@wordpress/scripts](https://github.com/WordPress/gutenberg/tree/HEAD/packages/scripts) from 30.16.0 to 30.17.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/scripts/CHANGELOG.md)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/scripts@30.17.0/packages/scripts)

---
updated-dependencies:
- dependency-name: "@wordpress/scripts"
  dependency-version: 30.17.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
NPM Dev(deps-dev): Bump postcss from 8.5.3 to 8.5.4

Bumps [postcss](https://github.com/postcss/postcss) from 8.5.3 to 8.5.4.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.5.3...8.5.4)

---
updated-dependencies:
- dependency-name: postcss
  dependency-version: 8.5.4
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

arunshenoy99 reviewed (commented):
@AleTorrisi Could we update this branch with the base branch `enhance/v1-via-huapi`?

arunshenoy99 committed:
Merge pull request #69 from newfold-labs/dependabot/npm_and_yarn/main/wordpress/scripts-30.17.0

NPM Dev(deps-dev): Bump @wordpress/scripts from 30.16.0 to 30.17.0

arunshenoy99 committed:
Merge pull request #72 from newfold-labs/dependabot/npm_and_yarn/main/postcss-8.5.4

NPM Dev(deps-dev): Bump postcss from 8.5.3 to 8.5.4

arunshenoy99 committed:
Add CODEOWNERS

arunshenoy99 committed:
Merge pull request #73 from newfold-labs/add/codeowners

Add CODEOWNERS

AleTorrisi committed:
Merge commit '5f706b274dd3c9a23520327860dc00f20ceb0b8e' into tweak/PRESS7-409

# Conflicts:

AleTorrisi committed:
Merge branch 'main' into tweak/PRESS7-409

# Conflicts:
#	package-lock.json
#	package.json

AleTorrisi committed:
Fix: failure on cypress test

arunshenoy99 reviewed (changes requested):
@AleTorrisi let's resolve the merge conflict and remove the build changes from this PR. Looks like there are a lot of file changes unrelated to the PHP version. Let's keep only the necessary new changes. Thanks!
